controllers/secretagentconfiguration_controller.go

```diff
@@ -229,6 +243,14 @@ func (reconciler *SecretAgentConfigurationReconciler) updateStatus(ctx context.C
 	instance.Status.ManagedK8sSecrets = secretNames
 	instance.Status.TotalManagedObjects = totalManagedObjects
 
+	if errorFound {
+		instance.Status.State = v1alpha1.SecretAgentConfigurationError
+	} else if inProgress {
+		instance.Status.State = v1alpha1.SecretAgentConfigurationInProgress
+	} else {
```

Comment:
We default `errorFound` to false, if we miss catching some error or whatever in the reconcile loop this leads to a default state of "completed". I'd rather have an unknown or unhandled state as a safety, I think the "complete" state might lead to false positives.

Reply:
```suggestion
	} else if !errorFound && !inProggress {
			instance.Status.State = v1alpha1.SecretAgentConfigurationInProgress
	} else {
	instance.Status.State = v1alpah1.SecretAgentConfigurationMisconfigured
	}
```

Reply:
we decided to keep it as is